apps/sv/frontend/src/components/forms/GrantRevokeFeaturedAppForm.tsx

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ActionRequiringConfirmation } from '@daml.js/splice-dso-governance/lib/Splice/DsoRules';
+import { CommonFormData } from './CreateProposalform';
+import { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useDsoInfos } from '../../contexts/SvContext';
+import dayjs from 'dayjs';
+import { createProposalActions, getInitialExpiration } from '../../utils/governance';
+import { dateTimeFormatISO } from '@lfdecentralizedtrust/splice-common-frontend-utils';
+import { useAppForm } from '../../hooks/form';
+import { SupportedActionTag } from '../../utils/types';
+import { ContractId } from '@daml/types';
+import { FeaturedAppRight } from '@daml.js/splice-amulet/lib/Splice/Amulet';
+import {
+  validateEffectiveDate,
+  validateExpiration,
+  validateExpiryEffectiveDate,
+  validateGrantRevokeFeaturedAppRight,
+  validateSummary,
+  validateUrl,
+} from './formValidators';
+import { FormLayout } from './FormLayout';
+import { Alert, Box, FormControlLabel, Radio, RadioGroup, Typography } from '@mui/material';
+
+interface ExtraFormField {
+  // represents appId for granting and contractId for revoking
+  idValue: string;
+}
+
+type GrantRevokeFeaturedAppFormData = CommonFormData & ExtraFormField;
+
+export interface GrantRevokeFeaturedAppFormProps {
+  selectedAction: GrantRevokeFeaturedAppActions;
+  onSubmit: (
+    data: GrantRevokeFeaturedAppFormData,
+    action: ActionRequiringConfirmation
+  ) => Promise<void>;
+}
+
+export type GrantRevokeFeaturedAppActions = Extract<
```

Comment:
I've so far had separate forms but in this case, it made sense to combine both into one dynamic form